script/gn-asar.js

```diff
@@ -41,23 +41,15 @@ try {
   // Copy all files to a tmp dir to avoid including scrap files in the ASAR
   for (const file of files) {
     const newLocation = path.resolve(tmpPath, path.relative(base[0], file));
-    fs.mkdirsSync(path.dirname(newLocation));
+    fs.mkdirSync(path.dirname(newLocation), { force: true, recursive: true });
     fs.writeFileSync(newLocation, fs.readFileSync(file));
   }
+
+  // Create the ASAR archive
+  asar.createPackageWithOptions(tmpPath, out[0], {});
 } catch (err) {
   console.error('Unexpected error while generating ASAR', err);
-  fs.remove(tmpPath)
-    .then(() => process.exit(1))
-    .catch(() => process.exit(1));
-  return;
+  process.exitCode = 1;
+} finally {
+  fs.rmSync(tmpPath, { force: true, recursive: true });
```

Comment:
fs.rmSync errors are not being caught here
```suggestion
 fs.promises.(tmpPath, { force: true, recursive: true })
     .then(() => process.exit(1))
    .catch(() => process.exit(1));
 return
```

Reply:
They're not caught on line 62 of the original either